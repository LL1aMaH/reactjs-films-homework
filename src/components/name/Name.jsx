import React from 'react';
import PropTypes from 'prop-types';
import './Name.scss';

const Name = ({ dataFilm, genreFilm }) => (
  <div>
    {dataFilm && (
      <div className="name">
        <div className="title">{dataFilm.title}</div>
        <div className="info">
          {genreFilm.map((result) => (
            <div key={result} className="filmGenre">
              {result}
            </div>
          ))}
        </div>
        <div className="stars">
          <span>{dataFilm.vote_average}</span>
        </div>
      </div>
    )}
  </div>
);
Name.propTypes = {
  dataFilm: PropTypes.shape({
    title: PropTypes.string,
    vote_average: PropTypes.number,
  }),
};

Name.defaultProps = {
  dataFilm: {},
  genreFilm: [],
};

export default Name;
