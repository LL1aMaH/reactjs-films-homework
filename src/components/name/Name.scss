.title {
  color: white;
  font-size: 35px;
  font-weight: 600;
  margin-bottom: 0;
}

.name {
  display: flex;
  flex-direction: column;
}

.info {
  display: flex;
  color: white;
  font-size: 25px;
}

.stars{
  color: white;
  font-size: 20px;
}

.filmGenre {
  padding-right: 15px;
}
