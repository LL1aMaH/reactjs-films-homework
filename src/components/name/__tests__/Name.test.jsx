import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import Name from '../Name';

describe('<Name />', () => {
  it('<Name />', () => {
    const data = {
      name: 'Gentleman',
      genre: 'action',
      time: '113 min',
      stars: 10,
    };
    const wrapper = shallow(<Name dataFilm={data} />);
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
