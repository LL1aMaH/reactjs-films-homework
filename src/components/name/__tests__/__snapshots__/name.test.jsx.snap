// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Name /> <Name /> 1`] = `
<div
  className="name"
>
  <h2>
    Gentleman
  </h2>
  <div
    className="info"
  >
    <span>
      action
    </span>
    <div
      className="vl"
    />
    <span>
      113 min
    </span>
  </div>
  <div
    className="stars"
  >
    <span>
      10
    </span>
  </div>
</div>
`;
